nomad/structs/node_test.go

```diff
@@ -9,11 +9,74 @@ import (
 	"time"
 
 	"github.com/go-jose/go-jose/v3/jwt"
+	"github.com/hashicorp/go-version"
 	"github.com/hashicorp/nomad/ci"
 	"github.com/shoenig/test/must"
 	"github.com/stretchr/testify/require"
 )
 
+func TestNode_meetsMinimumVersion(t *testing.T) {
```

Comment:
I know you've basically copied the server logic here but can we toss in a `+ent` test case just to satisfy my paranoia about shipping this without it working in ENT?

Reply:
Oh absolutely, good shout. I duplicated the CE, so we have enterprise variations here: https://github.com/hashicorp/nomad/pull/26842/commits/ad23aa2661be40af511732c6e270dd2acd565e0c

I checked the node attribute locally to make sure I got the right style:
```console
$ nomad node status -verbose  b4cc68c5 |grep nomad.version
nomad.version                     = 1.10.6-dev+ent
```